File: src/main/java/com/vjiki/music/dto/ParticipantResponse.java
package com.vjiki.music.dto;

import java.time.OffsetDateTime;
import java.util.UUID;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ParticipantResponse {
    private UUID userId;
    private String userEmail;
    private String userNickname;
    private String userAvatarUrl;
    private String role;
    private OffsetDateTime joinedAt;

    @JsonProperty("isMuted")
    private Boolean isMuted;
}

File: src/main/kotlin/com/vjiki/music/dto/ParticipantResponse.kt
package com.vjiki.music.dto

import java.time.OffsetDateTime
import java.util.UUID

data class ParticipantResponse(
    val userId: UUID,
    val userEmail: String,
    val userNickname: String,
    val userAvatarUrl: String?,
    val role: String,
    val joinedAt: OffsetDateTime?,
    val isMuted: Boolean
)

